src/workflow/Util.cpp

```diff
@@ -203,30 +187,29 @@ bool addEnergyPlusOutputRequest(Workspace& workspace, IdfObject& idfObject) {
     workspace.addObject(idfObject);
 
     return true;
-  }
-
-  //  static const std::vector<IddObjectType> allowedUniqueObjects{
-  //    // IddObjectType::Output_EnergyManagementSystem, // TODO: have to merge
-  //    // IddObjectType::OutputControl_SurfaceColorScheme, // TODO: have to merge
-  //    IddObjectType::Output_Table_SummaryReports,  // TODO: have to merge
-  //
-  //    // Not allowed
-  //    // IddObjectType::OutputControl_Table_Style,
-  //    // IddObjectType::OutputControl_ReportingTolerances,
-  //    // IddObjectType::Output_SQLite,
-  //  };
-
-  if (iddObjectType == IddObjectType::Output_Table_SummaryReports) {
+  } else if (iddObjectType == IddObjectType::Output_Table_SummaryReports) {
     auto summaryReports = workspace.getObjectsByType(iddObjectType);
     if (summaryReports.empty()) {
       workspace.addObject(idfObject);
       return true;
     } else {
-      mergeOutputTableSummaryReports(summaryReports.front(), idfObject);
+      return mergeOutputTableSummaryReports(summaryReports.front(), idfObject);
     }
+  } else {
+    // It's a unique one, for which we didn't write a merge, so remove it first
+    auto existingUniqueObjects = workspace.getObjectsByType(iddObjectType);
+    if (!existingUniqueObjects.empty()) {
+      // Technically it should be size == 1
+      if (std::ranges::any_of(existingUniqueObjects, [&idfObject](const auto& wo) { return idfObject.dataFieldsEqual(wo); })) {
+        return false;
+      }
+      for (auto& wo : existingUniqueObjects) {
+        wo.remove();
+      }
+    }
+    workspace.addObject(idfObject);
+    return true;
```

Comment:
If it's a unique one, check if there's the same exact object in there already and return false if so.

Otherwise, remove it first before adding the one (=replace).

We shouldn't end up with 2 "Output:SQLite" objects for eg, or E+ will throw